lib/find-node-directory.js

```diff
@@ -0,0 +1,40 @@
+var path = require('path')
+  , log = require('npmlog')
+
+function findNodeDirectory() {
+  // Have a look to see what is above us, to try and work out where we are
+  npm_parent_directory = path.join(__dirname, '../../../..')
+  log.verbose('node-gyp root', 'npm_parent_directory is ' + path.basename(npm_parent_directory))
```

Comment:
Style nit: can you try to keep lines <= 80 columns?  I know node-gyp isn't very consistent in that respect but I'd like new code to not have overly long lines.